karapace/schema_reader.py

```diff
@@ -88,6 +88,43 @@ class SchemaType(str, Enum):
     PROTOBUF = "PROTOBUF"
 
 
+class OffsetsWatcher:
+    """Synchronization container for threads to wait until an offset is seen.
+
+    This works under the assumption offsets are used only once, which should be
+    correct as long as no unclean leader election is performed.
+    """
+
+    def __init__(self) -> None:
+        # Lock used to protected _events, any modifications to that object must
+        # be performed with this lock acquired
+        self._events_lock = Lock()
+        self._events: Dict[Offset, Event] = dict()
+
+    def offset_seen(self, new_offset: int) -> None:
+        with self._events_lock:
+            # Note: The reader thread could be already waiting on the offset,
+            # in that case the existing object must be used, otherwise a new
+            # one is created.
+            event = self._events.setdefault(new_offset, Event())
+            event.set()
+
+    def wait_for_offset(self, expected_offset: int, timeout: int) -> bool:
```

Comment:
`event.wait` takes timeout as a floating point number specifying a timeout for the operation in seconds (or fractions thereof). Therefore suggest convering type to float, and somehow documenting the unit of timeout (seconds).